### PR TITLE
Add hide completed cards button

### PR DESCRIPTION
**Pull request by AderbalFilho**
Shown:
image

Hidden:
image

### CODE REVIEW

File: src/app.vue [hunk 1]
```diff
@@ -702,6 +715,12 @@ export default {
         alert("Couldn't load file's content");
       };
     },
+    hideCompleted() {
```
Comment (vhoyer) [inline src/app.vue:718]:
```suggestion
    toggleCompleted() {
```

File: src/app.vue [hunk 2]
```diff
@@ -71,28 +78,30 @@
...
-          @edit-title="editField($event, card.id)"
-          @edit-project="editField($event, card.id)"
-          @edit-description="editField($event, card.id)"
-          @edit-time="editField($event, card.id)"
-          @edit-eta="editField($event, card.id)"
-          @contextmenu="contextmenu($event, card.id)"
-        />
+        <template v-for="card in cards">
+          <card
+            v-if="showCard(card)"
```
[the remark this answers is not in the document]
Comment (vhoyer) [reply src/app.vue:83]: actually there is a thing I like to do with those kind of lists that is the following:

```js
function cardList() {
  const filters = []

  if (this.hideCompleted) {
    filters.push((card) => {
      return this.taskStates[card.taskState] !== 'done'
    })
  }

  return filters.reduce((cardList, handler) => {
    return cardList.filter(handler)
  }, this.cards)
}
```

this way we are also are preparing the ground to add other kinds of transformations while keeping the implamentation simple :D

File: src/app.vue (hunk 1 above)
Comment (AderbalFilho) [reply src/app.vue]: Done!

File: src/app.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (AderbalFilho) [reply src/app.vue]: I've done as you said in the second comment, =D

**Comment by vhoyer** [comment]
nice :D

**Review by vhoyer — APPROVED** [review]

**Pull request merged**